src/libcore/slice.rs

```diff
@@ -1558,8 +1558,18 @@ impl<T: Eq> Eq for [T] {}
 
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<T: Ord> Ord for [T] {
+    #[inline]
     fn cmp(&self, other: &[T]) -> Ordering {
-        self.iter().cmp(other.iter())
+        let l = cmp::min(self.len(), other.len());
+
+        for i in 0..l {
+            match self[i].cmp(&other[i]) {
```

Comment:
Are bounds checks eliminated here? [This](https://users.rust-lang.org/t/how-to-zip-two-slices-efficiently/2048) suggests you need to slice and other to length `l` for it to be.